ui/imports/shared/popups/send/SendModal.qml

```diff
@@ -72,16 +72,22 @@ StatusDialog {
         popup.store.authenticateAndTransfer(d.uuid)
     }
 
-    property var recalculateRoutesAndFees: Backpressure.debounce(popup, 600, function() {
-        if(!!popup.selectedAccount && !!popup.selectedAccount.address && !!holdingSelector.selectedItem
-                && recipientInputLoader.ready && (amountToSend.ready || d.isCollectiblesTransfer)) {
-            popup.isLoading = true
-            d.routerError = ""
-            d.routerErrorDetails = ""
+    property var debounceRecalculateRoutesAndFees: Backpressure.debounce(popup, 1000, function() {
+        if(d.areInputParametersValid()) {
             popup.store.suggestedRoutes(d.uuid, d.isCollectiblesTransfer ? "1" : amountToSend.amount, "0", d.extraParamsJson)
         }
     })
 
+    function recalculateRoutesAndFees() {
+        if(d.areInputParametersValid()) {
+            popup.isLoading = true
+        }
+        d.uuid = popup.generateUuid()
+        d.routerError = ""
+        d.routerErrorDetails = ""
+        debounceRecalculateRoutesAndFees()
+    }
+
     property var generateUuid: () => { return Utils.uuid() }
```

Comment:
why is this not an internal property?

Reply:
we don't really need this, we could be using `Utils.uuid()` directly, will remove it

Reply:
done